### ISSUE TITLE
Tests can't be run programmatically more than once

### ISSUE THREAD

**Issue by fvictorio** [nomiclabs/hardhat#1720]
Reproduction steps:

1. Create the sample project
2. Create a script like this:

    ```js
    const hre = require("hardhat");

    async function main() {
      await hre.run("test")
      await hre.run("test")
    }

    main()
      .then(() => process.exit(0))
      .catch(error => {
        console.error(error);
        process.exit(1);
      });
    ```

3. Run `hh run script.js`

**Expected result**: Tests should be run twice.

**Actual result**: the second run doesn't find any tests:

```

  Greeter
Deploying a Greeter with greeting: Hello, world!
Changing greeting from 'Hello, world!' to 'Hola, mundo!'
    ✓ Should return the new greeting once it's changed (782ms)


  1 passing (785ms)



  0 passing (0ms)
```

This seems like a "mocha caching something" thing.

**Comment by alcuadrado** [comment]
> This seems like a "mocha caching something" thing.

It's mocha loading the test files with `require`, so they only get interpreted once.

**Comment by 0xww** [comment]
I found same issue closed on mocha repo [#2783](https://github.com/mochajs/mocha/issues/2783). can be helpful.

**Comment by fvictorio** [comment]
Reopening so we can close it when this is released.

**Comment by fvictorio** [comment]
Fixed in [`hardhat@2.8.1`](https://github.com/nomiclabs/hardhat/releases/tag/hardhat%402.8.1).
